js/app.js

```diff
@@ -1,39 +1,88 @@
-    // 1. zamień wartości przekazane przez parametr na typ number
-    // 2. sprawdź czy są one poprawne
-    // 3. jeśli tak to wykonaj działanie i zapisz jego resultat
-    // 4. dodaj do historii operacji to działanie w fomie: 1 + 1 = 2
-}
+Calculator.prototype.performOperation = function (
+  num1,
+  num2,
+  operator,
+  operationFn
+) {
+  num1 = Number(num1);
+  num2 = Number(num2);
+
+  if (isNaN(num1) || isNaN(num2)) {
+    alert('Musisz podać dwie liczby!');
+    return;
+  }
+
+  if (operator === '/' && num2 === 0) {
+    alert('Nie można dzielić przez 0!');
+    return;
+  }
+  const result = operationFn(num1, num2);
+  const operation = `${num1} ${operator} ${num2} = ${result}`;
+  this.history.push(operation);
+  alert(operation);
+};
+
+Calculator.prototype.add = function (num1, num2) {
+  this.performOperation(num1, num2, '+', (a, b) => a + b);
+};
+Calculator.prototype.subtract = function (num1, num2) {
+  this.performOperation(num1, num2, '-', (a, b) => a - b);
+};
 
+Calculator.prototype.multiply = function (num1, num2) {
+  this.performOperation(num1, num2, '*', (a, b) => a * b);
+};
+
+Calculator.prototype.divide = function (num1, num2) {
+  this.performOperation(num1, num2, '/', (a, b) => a / b);
+};
+
+Calculator.prototype.pow = function (num1, num2) {
+  this.performOperation(num1, num2, '^', (a, b) => {
+    let result = 1;
+    for (let i = 0; i < b; i++) {
+      result *= a;
+    }
+    return result;
+  });
+};
 const calc = new Calculator();
 let action, promptContent, isCorrectAction, number1, number2;
-do { 
-    promptContent = 'Podaj jaką operację chcesz wykonać (+, -, *, /, ^) i potwierdź. \n'; // \n - znak nowej linii
-    promptContent += 'Jeśli chcesz zrezygnować wciśnij Anuluj. \n';
-    promptContent += 'Lista poprzednich operacji: \n' + calc.getHistoryAsString();
-
-    action = prompt(promptContent);
-    isCorrectAction = calc.isCorrectAction(action);
-    if(isCorrectAction) {
-        number1 = prompt('Podaj liczbę nr 1');
-        number2 = prompt('Podaj liczbę nr 2');
-
-        if(action === '+') {
-            calc.add(number1, number2);
-        }
+do {
+  promptContent =
+    'Podaj jaką operację chcesz wykonać (+, -, *, /, ^) i potwierdź. \n'; // \n - znak nowej linii
+  promptContent += 'Jeśli chcesz zrezygnować wciśnij Anuluj. \n';
+  promptContent += 'Lista poprzednich operacji: \n' + calc.getHistoryAsString();
+
+  action = prompt(promptContent);
+  isCorrectAction = calc.isCorrectAction(action);
+  if (isCorrectAction) {
+    number1 = prompt('Podaj liczbę nr 1');
+    number2 = prompt('Podaj liczbę nr 2');
+
+    if (action === '+') {
+      calc.add(number1, number2);
+    } else if (action === '-') {
+      calc.subtract(number1, number2);
+    } else if (action === '*') {
+      calc.multiply(number1, number2);
+    } else if (action === '/') {
+      calc.divide(number1, number2);
+    } else if (action === '^') {
+      calc.pow(number1, number2);
     }
```

Comment:
Podobnie tutaj. Dodanie nowej operacji to kolejny "if" - przy dużej liczbie operacji arytmetycznych będzie to niepotrzebnie komplikować kod. Może spróbować innego rozwiązania: https://devmentor.pl/b/mniej-instrukcji-warunkowych